src/org/parosproxy/paros/network/SSLConnector.java

```diff
@@ -530,7 +531,8 @@ public Socket createSocket(Socket socket, String host, int port,
 	 */
 	public Socket createTunnelServerSocket(String targethost, Socket socket) throws IOException {
 		// ZAP: added host name parameter
-		SSLSocket s = (SSLSocket) getTunnelSSLSocketFactory(targethost).createSocket(socket, socket
+		InetAddress listeningAddress = socket.getInetAddress();

```

Comment:
This should call `getLocalAddress()`, otherwise we are generating the cert with remote address.